internal/storage/storageutil/client_test.go

```diff
@@ -47,54 +46,57 @@ func (t *clientTest) validateProxyInTransport(httpClient *http.Client) {
 
 // Tests
 
-func (t *clientTest) TestCreateTokenSrcWithCustomEndpoint() {
-	url, err := url.Parse(CustomEndpoint)
-	AssertEq(nil, err)
-	sc := GetDefaultStorageClientConfig()
-	sc.CustomEndpoint = url
+func (t *clientTest) TestCreateHttpClientWithHttp1() {
+	sc := GetDefaultStorageClientConfig() // By default http1 enabled
 
-	tokenSrc, err := CreateTokenSource(&sc)
+	httpClient, err := CreateHttpClient(&sc)
 
 	ExpectEq(nil, err)
-	ExpectNe(nil, &tokenSrc)
+	ExpectNe(nil, httpClient)
+	ExpectEq(sc.HttpClientTimeout, httpClient.Timeout)
 }
 
-func (t *clientTest) TestCreateTokenSrcWhenCustomEndpointIsNil() {
+func (t *clientTest) TestCreateHttpClientWithHttp2() {
```

Comment:
We should also write unit test for:

(a) CreateHttpClientWithHttp2 (with and without auth)
(b) CreateHttpClientWithHttp1 (with and without auth)